client/state/ui/language/actions.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import switchLocale from 'lib/i18n-utils/switch-locale';
+import i18n from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import {
+	LOCALE_SET,
+} from 'state/action-types';
+
+/**
+ * Set the ui locale
+ *
+ * @param   {Object} localeSlug the locale slug to change the locale to
+ * @returns {Function} Action thunk
+ */
+export const setLocale = localeSlug => {
+	switchLocale( localeSlug );
```

Comment:
SSR ignores everything that's not the default locale:
https://github.com/Automattic/wp-calypso/blob/master/server/render/index.js#L95

Also `i18n` lib is a singleton, so it's state will be shared between all the requests,
calling `i18n.setLocale` on server side won't get us the result we want.

So I'm not sure if we even need to call it on the server.